peer/network_test.go

```diff
@@ -361,6 +365,50 @@ func TestGossip(t *testing.T) {
 	assert.True(t, gossipHandler.received)
 }
 
+func TestCrossChainRequest(t *testing.T) {
+	var net Network
+	codecManager := buildCodec(t, TestMessage{})
+	crossChainCodecManager := buildCodec(t, ExampleCrossChainRequest{}, ExampleCrossChainResponse{})
+
+	sender := testAppSender{
+		sendCrossChainAppRequestFn: func(requestingChainID ids.ID, requestID uint32, requestBytes []byte) error {
+			// Because CrossChainAppRequest is not implemented in Subnet-EVM, we cannot mimic it in the tests.
+			// We assume it works in the other chain we send the CCR and mock the response.
+			go func() {
+				responseBytes, err := crossChainCodecManager.Marshal(message.Version, ExampleCrossChainResponse{Response: "this is an example response"})
+				assert.NoError(t, err)
+
+				if err := net.CrossChainAppResponse(context.Background(), requestingChainID, requestID, responseBytes); err != nil {
+					panic(err)
+				}
+			}()
+			return nil
+		},
+	}
+
+	net = NewNetwork(sender, codecManager, ids.EmptyNodeID, 1)
+
+	net.SetCrossChainRequestHandler(nil)
+	client := NewNetworkClient(net)
+
+	exampleCrossChainRequest := ExampleCrossChainRequest{
+		Message: "hello this is an example request",
+	}
+
+	crossChainRequest, err := buildCrossChainRequest(crossChainCodecManager, exampleCrossChainRequest)
+	assert.NoError(t, err)
+
+	chainID := ids.ID(ethcommon.BytesToHash([]byte{1, 2, 3, 4, 5}))
+	responseBytes, err := client.SendCrossChainRequest(chainID, crossChainRequest)
+	assert.NoError(t, err)
+
+	var response ExampleCrossChainResponse
+	if _, err = crossChainCodecManager.Unmarshal(responseBytes, &response); err != nil {
+		t.Fatal("unexpected error during unmarshal", err)
+	}
+	assert.Equal(t, "this is an example response", response.Response)
```

Comment:
could we update the test to use `require`?

Reply:
Sure! is there a benefit?  Just asking for knowledge. I'm using assert to be uniform with the other tests.

Reply:
The primary benefit IMO is avoiding the verbosity of:
```
if err := ...; err != nil {
 t.Fatal(...)
}
```
instead we can write:
```
err := ...
require.NoError(t, err)
```
or
```
require.NoError(t, ...)
```

Using require is preferred for new tests.